packages/dendron-cli/src/commands/base.ts

```diff
@@ -102,6 +106,51 @@ export abstract class CLICommand<
     this.L.info({ msg: `Telemetry is disabled? ${segment.hasOptedOut}` });
   }
 
+  async validateConfig(opts: { wsRoot: string }) {
+    const { wsRoot } = opts;
+    const ws = new WorkspaceService({ wsRoot });
+    const dendronConfig = ws.config;
+    const configVersion = ConfigUtils.getProp(dendronConfig, "version");
```

Comment:
if the `version` doesn't exist, we currently assume that it equals 4. can this get us into trouble?

Reply:
That's right. would need to look at the raw value and not assume a default with `getProp`.
For now I think it makes sense to stop all execution if the `version` field doesn't exist.
Later down the line once we introduce json schema validation, we will have to take snapshots of every version of the JSON schema and use that to determine the real `version`